### DISCUSSION TITLE
Fedora is too unstable for Servers - Use CentOS Stream?

### DISCUSSION THREAD

**Discussion by driverpt** [amazonlinux/amazon-linux-2022#37] [Ideas]
After seeing your announcement, not that i don't like Fedora, but Fedora is more suited for Desktop and daily usage, i find it too Unstable for Server environment. 

I suggest switching to CentOS Stream which is more Stable for Servers.

**Comment by stewartsmith** [comment]
Amazon Linux 2022 (and beyond) aren't direct rebuilds of Fedora, we base on Fedora and modify it as needed to solve our customer problems (such as a longer life cycle for a particular release).

What specifically do you find unstable about Fedora? Is it particular bugs? ABI compatibility between releases? Behaviour changes between releases?

**Reply by driverpt** [reply]
Fedora is aimed for Desktop, which mean shorter release cycles and more Unstable (let's put it like this) packages. From my experience, in Servers you want to keep the environment as Stable as possible, which is why i strongly believe that CentOS is a much more appropriate and Stabler version for Servers.

I'm just giving a heads-up that if you use Fedora, it's going to be costly for AWS to maintain it. But from a "Customer" Perspective as long as it works. (e.g.: Ubuntu is forked of Debian's `testing` Branch)

**Reply by jaltman** [reply]
> Amazon Linux 2022 (and beyond) aren't direct rebuilds of Fedora, we base on Fedora and modify it as needed to solve our customer problems (such as a longer life cycle for a particular release).
> 
> What specifically do you find unstable about Fedora? Is it particular bugs? ABI compatibility between releases? Behaviour changes between releases?

btrfs for starters.  

Prior Amazon Linux distros were based on rhel.  The new upstream for rhel is CentOS Stream. 

While I want kafs and rxrpc which are enabled by default in Fedora to be in Amazon Linux 2022 for use with Kubernetes which is disabled in rhel, I agree that CentOS Stream is a better choice for a base especially if Amazon Linux wishes enterprise and scientific workloads to migrate to Amazon Linux 2022.

**Reply by stewartsmith** [reply]
Amazon Linux 2022 AMIs have XFS as the root file system, like AL2 before it.

**Reply by driverpt** [reply]
@stewartsmith , it's still a huge mistake by AWS to Fork from Fedora

**Reply by stewartsmith** [reply]
I'm asking about specific concerns as it's also true that Fedora is the base for RHEL and thus also CentOS, so I am unclear as to what specific problems you've observed with Fedora and how one downstream distribution largely derived from Fedora would be affected by them and another would not.

**Reply by jaltman** [reply]
> I'm asking about specific concerns as it's also true that Fedora is the base for RHEL and thus also CentOS, so I am unclear as to what specific problems you've observed with Fedora and how one downstream distribution largely derived from Fedora would be affected by them and another would not. 

It is true that Fedora used to be the base for RHEL but that is no longer true. The current flow is Fedora to CentOS Stream to RHEL. 

The benefit of CentOS Stream is that it provides the community the ability contribute changes that flow directly into minor RHEL updates.  Unlike old CentOS or Alma or Rocky which provide no path for community investment.  Facebook, CERN, Fermilab and others are active contributors to CentOS Stream. There are also active SIGs in the CentOS Stream community.

**Reply by driverpt** [reply]
Currently i'd risk saying that Fedora is on the same level as Arch Linux, which means Bleeding Edge Packages.

I don't want a package upgrade messing up my deployments in EC2 (Installed via ASG + EC2 Metadata Script), which is much more likely to happen in Fedora.

**Reply by stewartsmith** [reply]
We're very conscious about package updates and their impact on customers deployments. There's two ways we're addressing this in AL2022: 1) continuing our care in choosing what to update in Amazon Linux and when (like we do with AL2 today) and 2) the version locking feature of AL2022 means you *never* get updates without *explicitly* taking action to update the version you're locked to.

**Reply by crabby345** [reply]
You're not adding Fedora repos directly (or not intended to) so AWS are just taking control of the filter through and deciding when a Fedora package is stable enough to be incorporated.

I imagine a lot of the decisions will parallel CentOS Stream/Rhel/Rocky, but with flexibility to diversify. I thought AL2 was already a hybrid combination of CentOS and direct Fedora.
